src/main/java/com/couchbase/lite/store/SQLiteViewStore.java

```diff
@@ -250,48 +255,115 @@ public void updateIndex() throws CouchbaseLiteException {
         }
 
         store.beginTransaction();
-        Status result = new Status(Status.INTERNAL_SERVER_ERROR);
+        boolean success = false;
         Cursor cursor = null;
-
         try {
-            long last = getLastSequenceIndexed();
-            long dbMaxSequence = store.getLastSequence();
+            // If the view the update is for doesn't need any update, don't do anything:
+            final long dbMaxSequence = store.getLastSequence();
+            final long forViewLastSequence = getLastSequenceIndexed();
+            if (forViewLastSequence >= dbMaxSequence) {
+                success = true;
+                return new Status(Status.NOT_MODIFIED);
+            }
+
+            // Check whether we need to update at all,
+            // and remove obsolete emitted results from the 'maps' table:
             long minLastSequence = dbMaxSequence;
+            long[] viewLastSequence = new long[inputViews.size()];
+            int deletedCount = 0;
+            int i = 0;
+            HashSet<String> docTypes = new HashSet<String>();
+            HashMap<String, String> viewDocTypes = null;
+            boolean allDocTypes = false;
+            HashMap<Integer, Integer> viewTotalRows = new HashMap<Integer, Integer>();
+            ArrayList<SQLiteViewStore> views = new ArrayList<SQLiteViewStore>();
+            ArrayList<Mapper> mapBlocks = new ArrayList<Mapper>();
+
+            for (ViewStore v : inputViews) {
+                SQLiteViewStore view = (SQLiteViewStore)v;
+                ViewStoreDelegate delegate = view.getDelegate();
+                Mapper map = delegate.getMap();
+                if (map == null) {
+                    if (view == this) {
+                        String msg = String.format("Cannot index view %s: " +
```

Comment:
After thought about this, I prefer to not check if view is null as 
1. To let it failed as it means that we have a more serious bug somewhere else.
2. The case shouldn't be happening.